gh.go

```diff
@@ -107,6 +108,35 @@ func GQLClient(opts *api.ClientOptions) (api.GQLClient, error) {
 	return iapi.NewGQLClient(opts.Host, opts), nil
 }
 
+// HTTPClient builds a client that can be passed to another library.
+// As part of the configuration a hostname, auth token, and default set of headers are resolved
+// from the gh environment configuration. These behaviors can be overridden using the opts argume
+func HTTPClient(opts *api.ClientOptions) (*http.Client, error) {
+	var cfg config.Config
+	var token string
+	var err error
+	if opts == nil {
+		opts = &api.ClientOptions{}
+	}
+	if opts.Host == "" || opts.AuthToken == "" {
+		cfg, err = config.Load()
+		if err != nil {
+			return nil, err
+		}
+	}
+	if opts.Host == "" {
+		opts.Host = cfg.Host()
+	}
+	if opts.AuthToken == "" {
+		token, err = cfg.AuthToken(opts.Host)
+		if err != nil {
+			return nil, err
+		}
+		opts.AuthToken = token
```

Comment:
This `http.Client` instance will be configured with a `github.com` authentication token. However, a `http.Client` instance can be used to make arbitrary requests to any host, and we don't know how it will be used. We absolutely would *not* want that the `github.com` auth token is sent to non-`github.com` hosts. https://github.com/cli/go-gh/blob/6722e5635158d5b607f7d955253e826f7c87b5a4/internal/api/http.go#L219

If we were to expose a generic `http.Client`, we'd have to take precaution so that the token isn't set to other hosts. It might be as simple as only sending the authentication token with requests to hostnames that match `opts.Host`. What do you think?

Reply:
Yeah this sounds sensible to me. I'll make a start by adding something that stops the auth token being forward to non `opts.Host` requests as you suggest.

Thanks!

Reply:
@mislav I've made a small change that stops the authorization header from being included if the hosts don't match.

The interesting thing was that the configured host and request host will never be an explicit match (github.com vs api.github.com) so I am checking for existence of the one in the other.

Unsure if this is a viable solution so your feedback is very welcome!

Reply:
I've added some tests for the above change.  👍